lib/mods/replace-constructors.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
+const nodeTypes = require('ts-morph').TypeGuards;
+const globalsSet = require('../globals-set.js');
+
 /**
- * Checks if the given constructor name is global
+ * @param {Node} node - a typed AST Node
  *
- * @param {String} constructor - the name of the constructor
+ * @return {Boolean} whether the type of the node is from TypeScript definitions
+ */
+function nodeInTSDefs(node) {
+  return node.getType().compilerType.symbol
+      .getDeclarations()[0].getSourceFile().path
+      .includes('/node_modules/typescript');
+}
+
+/**
+ * @param {Node} node - a typed AST Node
+ *
+ * @return {Boolean} whether the type of the node is an IDL in a web spec
+ */
+function nodeInIDLSpecs(node) {
+  return globalsSet.has(node.getType()
+      .getSymbol().getName());
```

Comment:
Nit: I think this would look nicer with `node.getType()` on the second line too.